src/test/codewhisperer/service/securityIssueHoverProvider.test.ts

```diff
@@ -99,5 +102,81 @@ describe('securityIssueHoverProvider', () => {
             const actual = securityIssueHoverProvider.provideHover(mockDocument, new vscode.Position(2, 0), token.token)
             assert.strictEqual(actual.contents.length, 0)
         })
+
+        it('should remove hover if document changed on the same line', () => {
+            const securityIssueHoverProvider = new SecurityIssueHoverProvider()
+            const mockDocument = createMockDocument('def two_sum(nums, target):\nfor', 'test.py', 'python')
+            securityIssueHoverProvider.issues = [
+                {
+                    filePath: mockDocument.fileName,
+                    issues: [makeIssue({ startLine: 0, endLine: 1 })],
+                },
+            ]
+            const token = new vscode.CancellationTokenSource()
+            let actual = securityIssueHoverProvider.provideHover(mockDocument, new vscode.Position(0, 0), token.token)
+            assert.strictEqual(actual.contents.length, 1)
+
+            const changeEvent = createTextDocumentChangeEvent(
+                mockDocument,
+                new vscode.Range(new vscode.Position(0, 0), new vscode.Position(0, 1)),
+                '\n'
+            )
+            securityIssueHoverProvider.handleDocumentChange(changeEvent)
+
+            actual = securityIssueHoverProvider.provideHover(mockDocument, new vscode.Position(0, 0), token.token)
+            assert.strictEqual(actual.contents.length, 0)
+        })
+
+        it('should offset hover if document changed on a line above it', () => {
+            const securityIssueHoverProvider = new SecurityIssueHoverProvider()
+            const mockDocument = createMockDocument('def two_sum(nums, target):\nfor', 'test.py', 'python')
+            securityIssueHoverProvider.issues = [
+                {
+                    filePath: mockDocument.fileName,
+                    issues: [makeIssue({ startLine: 1, endLine: 2 })],
+                },
+            ]
+            const token = new vscode.CancellationTokenSource()
+            let actual = securityIssueHoverProvider.provideHover(mockDocument, new vscode.Position(1, 0), token.token)
+            assert.strictEqual(actual.contents.length, 1)
+
+            const changeEvent = createTextDocumentChangeEvent(
+                mockDocument,
+                new vscode.Range(new vscode.Position(0, 0), new vscode.Position(0, 1)),
```

Comment:
Does this range mean that the `d` in `def two_sum...` is replaced with a `\n`. Should it just be inserted?
```suggestion
                new vscode.Range(new vscode.Position(0, 0), new vscode.Position(0, 0)),
```